hadoop-project/pom.xml

```diff
@@ -204,6 +204,7 @@
     <assertj.version>3.12.2</assertj.version>
     <jline.version>3.9.0</jline.version>
     <powermock.version>1.5.6</powermock.version>
```

Comment:
Hey @bilaharith  @steveloughran ,  I'm still not sure about this change. Simply updating the existing "`powermock.version`" will cause build issues for some other projects, so new version variable "`powermock-api.version`" is introduced.
Some other projects already override "`powermock.version`" for `powermock-api-mockito2 `, such as Hadoop-Yarn [here](https://github.com/apache/hadoop/blob/4a70a0d81601f20ba8cecb37fae12b6a8be327b4/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-timelineservice-documentstore/pom.xml#L61). 
Maybe we can override its version in hadoop-azure's pom file too, instead of introducing the new version variable?
Other changes look good to me.

Reply:
Removing powermock for now, will raise another PR for powermock upgrade and then one for the mocktest